util/metrics/gauge.go

```diff
@@ -139,26 +142,11 @@ func (cv *gaugeValues) createFormattedLabel() {
 	cv.formattedLabels = buf.String()[1:]
 }
 
-// filterExpiredMetrics scans the gauge.valuesIndices map and removing all the gauges that
-// haven't been updated in the past <maxMetricRetensionDuration> units of time.
-func (gauge *Gauge) filterExpiredMetrics() {
```

Comment:
Removing this would be a problem if someone decides to use the label feature. It looks like that allows for creating dynamic guages based on user-defined keys. We don't currently use the dynamic labels feature in the networking code. In theory, someone might add a gauge by session ID (i.e. ping per connection). At that point removing this function would become be issue since it would grow unbounded.

A couple of possible solutions:
1) Remove the `labels` feature (it seems to only be used by `segment.go:EnterSegment` and that function is not called anywhere).
2) Return from this function immediately when `len(valueIndices) <= 1`
3) Set `maxMetricRetensionDuration` to something very large in `config.json`

Reply:
`MakeSegement` `EnterSegment` `LeaveSegment` never occur in the code. It's completely unused and I'm tempted to delete segment.go entirely.
`labels` is also unused, I checked all uses of Gauge.Set() and labels is always nil.

Since there's no problem with the code as it is but only as it might be, I'd like to finish this bugfix and file a new issue for further util/metrics cleanup of dead code.